### PR TITLE
Add support for Filesystem credentials.

### LINKED ISSUE #30: ProviderConfig doesn't support Filesystem credentials
### What happened?

The ProviderConfig for argocd has fields under `spec.credentials` for `env`, `fs`, and `secretRef`; but `fs` isn't supported.

### How can we reproduce it?

```yaml
apiVersion: argocd.crossplane.io/v1alpha1
kind: ProviderConfig
metadata:
  name: default
spec:
  credentials:
    source: Environment
    fs:
      path: /vault/secret/authToken
  serverAddr: https://argocd.example.com/
```

On apply:
```
The ProviderConfig "default" is invalid: spec.credentials.source: Unsupported value: "Filesystem": supported values: "None", "Secret", "Environment"
```

### What environment did it happen in?
Crossplane version: v1.9.1
Crossplane Provider argocd version: v0.1.0

I think this is just a one-line change at https://github.com/crossplane-contrib/provider-argocd/blob/main/package/crds/argocd.crossplane.io_providerconfigs.yaml#L98, but I wasn't able to get a build environment working in MacOS to actually make a PR.

### PR DESCRIPTION
**Pull request by cjyar**
### Description of your changes

Fixes #30

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.
        But it didn't work.

```
13:56:16 [ .. ] golangci-lint
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt

goroutine 1 [running]:
github.com/go-critic/go-critic/checkers.init.22()
        github.com/go-critic/go-critic@v0.6.3/checkers/embedded_rules.go:47 +0x4b4
13:56:23 [FAIL]
```

### How has this code been tested

I tested it in my own environment, where it was able to receive an ArgoCD token from Vault, authenticate to the ArgoCD server, and create an ArgoCD Cluster. This is the ControllerConfig I used:
```yaml
apiVersion: pkg.crossplane.io/v1alpha1
kind: ControllerConfig
metadata:
  name: argocd-vault-config
spec:
  metadata:
    annotations:
      vault.hashicorp.com/agent-inject: "true"
      vault.hashicorp.com/role: crossplane
      vault.hashicorp.com/agent-inject-secret-authtoken: secret/crossplane/argocd
      vault.hashicorp.com/agent-inject-template-authtoken: |
        {{- with secret "secret/crossplane/argocd" -}}
          {{ .Data.data.authToken }}
        {{- end -}}
      vault.hashicorp.com/ca-cert: /vault/tls/ca.crt
      vault.hashicorp.com/tls-secret: vault
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by cjyar** [comment]
When it tries to delete the Cluster, the provider crashes. I'm investigating; putting this stack trace here for safekeeping.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1aed90c]

goroutine 702 [running]:
github.com/crossplane-contrib/provider-argocd/pkg/controller/cluster.(*external).Observe(0xc000495ec0, {0x2335910, 0xc000305ce0}, {0x2345960?, 0xc00091ed80?})
	github.com/crossplane-contrib/provider-argocd/pkg/controller/cluster/controller.go:111 +0x28c
github.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile(0xc0007619a0, {0x2335948?, 0xc000a27380?}, {{{0x0?, 0x0?}, {0xc000a01530?, 0x14?}}})
	github.com/crossplane/crossplane-runtime@v0.15.0/pkg/reconciler/managed/reconciler.go:620 +0x17e3
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc000761a40, {0x2335948, 0xc000a27320}, {{{0x0?, 0x1e6c220?}, {0xc000a01530?, 0xc000bcc080?}}})
	sigs.k8s.io/controller-runtime@v0.10.3/pkg/internal/controller/controller.go:114 +0x222
...
```

File: pkg/clients/argocd.go [hunk 1]
```diff
@@ -67,6 +68,14 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		return nil, errors.Wrap(err, "cannot track ProviderConfig usage")
 	}
 
+	insecure := false
+	if pc.Spec.Insecure != nil {
+		insecure = *pc.Spec.Insecure
+	}
+	plaintext := false
+	if pc.Spec.Insecure != nil {
```
Comment (janwillies) [inline pkg/clients/argocd.go:76]:
```suggestion
	if pc.Spec.PlainText != nil {
```

**Review by janwillies — COMMENTED** [review]
Thanks @cjyar! One small nit, but looks good otherwise

**Comment by janwillies** [comment]
You can get around the linter issues when you add `// nolint:gocyclo`, e.g.
```suggestion
func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed) (*argocd.ClientOptions, error) { // nolint:gocyclo
```

File: pkg/clients/argocd.go (hunk 1 above)
Comment (janwillies) [reply pkg/clients/argocd.go:76]: The defaults should also be documented in https://github.com/crossplane-contrib/provider-argocd/blob/main/apis/v1alpha1/providerconfig_types.go#L30 and https://github.com/crossplane-contrib/provider-argocd/blob/main/apis/v1alpha1/providerconfig_types.go#L34

Comment (cjyar) [reply pkg/clients/argocd.go]: Ooh! Thanks for catching that.

**Comment by cjyar** [comment]
@janwillies I can't run golangci-lint from the Makefile, but I discovered I can run it manually. I fixed the lints that it found, but there might be some others that CI finds; PTAL.

**Comment by janwillies** [comment]
Thanks @cjyar! One last comment, could you please make a note about the new defaults in https://github.com/crossplane-contrib/provider-argocd/blob/main/apis/v1alpha1/providerconfig_types.go#L30 and https://github.com/crossplane-contrib/provider-argocd/blob/main/apis/v1alpha1/providerconfig_types.go#L34/? This makes sure they show up in the api documentation.
Then please squash the commits into one, sign it and we are good to go :)

**Comment by cjyar** [comment]
@janwillies I don't think I actually changed the defaults, since Go defaults `bool` to `false`. But you're right that the docs should be clear about it; I made the changes you suggested.

Squashed and signed.

**Comment by janwillies** [comment]
Thanks for the contribution @cjyar and sorry for the long wait!

**Pull request merged**